hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -34,6 +34,8 @@
 import os
 from os import path
 import yaml
+import zlib
+import traceback
```

Comment:
Are these imports already installed? Do we need to install them by specifying them in pyinstaller_requirements.txt?

Reply:
These libraries are part of core python libraries.

Reply:
I really thought I posted those change requests days and days ago, but they were pending (not showing)...

Reply:
@devagupt can more into depth on this issue, but the goal here is not to check file integrity - the goal is to know if the file we're continuing to send events from is the same file as before or has it been rotated and just happens to be again as big as the previous one. File integrity would not help as the file may grow between the checks.

Using a mechanism to validate parts of the content can also serve as indication of potential loss of collected data if Hubble was not the one that rotated it.

Reply:
@fossam It is correct that CRC mechanism is used to identify if we have missed sending any events to Splunk. If we found some deviations, then we are sending the whole log file again to ensure no events are missed. 
As 4.0 release deadline is approaching, we can pick up the changes proposed by @jettero in the future versions to use salt lib.

Reply:
@fossam It is correct that CRC mechanism is used to identify if we have missed sending any events to Splunk. If we found some deviations, then we are sending the whole log file again to ensure no events are missed.
As 4.0 release deadline is approaching, we can pick up the changes proposed by @jettero in the future versions to use salt lib.